### PR TITLE
tensorflow-lite: fix flatbuffers discovery + modernize

### PR DESCRIPTION
**Pull request by SpaceIm**
closes https://github.com/conan-io/conan-center-index/issues/10402

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/tensorflow-lite/all/conanfile.py [hunk 1]
```diff
@@ -96,37 +72,56 @@ def configure(self):
...
+        if self.options.shared:
+            if self.settings.os == "Linux" and not self.options["ruy"].shared:
+                raise ConanInvalidConfiguration(
+                        f"The project {self.name}/{self.version} with shared=True on Linux requires ruy:shared=True")
+
+    def build_requirements(self):
+        self.build_requires("ninja/1.10.2")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
```
Comment (AndreyMlashkin) [inline recipes/tensorflow-lite/all/conanfile.py:113]: Won't it fail, if there are no patches?

Comment (SpaceIm) [reply recipes/tensorflow-lite/all/conanfile.py:113]: Why do you think it would fail?

File: recipes/tensorflow-lite/all/conanfile.py [hunk 2]
```diff
@@ -32,26 +33,16 @@ class TensorflowLiteConan(ConanFile):
...
-    _cmake = None
 
-    @property
-    def _module_subfolder(self):
-        return os.path.join("lib", "cmake")
-
-    @property
-    def _module_file(self):
-        return "conan-official-{}-targets.cmake".format(self.name)
+    generators = "cmake", "cmake_find_package", "cmake_find_package_multi"
```
Comment (SpaceIm) [inline recipes/tensorflow-lite/all/conanfile.py:37]:
```suggestion

    short_paths = True
    generators = "cmake", "cmake_find_package", "cmake_find_package_multi"
```

File: recipes/tensorflow-lite/all/conanfile.py (hunk 1 above)
Comment (AndreyMlashkin) [reply recipes/tensorflow-lite/all/conanfile.py:115]: Sorry, it was wrong line, it should be ok

File: recipes/tensorflow-lite/all/conanfile.py [hunk 3]
```diff
@@ -32,26 +33,18 @@ class TensorflowLiteConan(ConanFile):
...
-    @property
-    def _module_file(self):
-        return "conan-official-{}-targets.cmake".format(self.name)
+    short_paths = True
+    generators = "cmake", "cmake_find_package", "cmake_find_package_multi"
 
     @property
     def _source_subfolder(self):
-        return "source_subfolder"
+        return "src"
```
Comment (AndreyMlashkin) [inline recipes/tensorflow-lite/all/conanfile.py:43]: Why do you need this change?

File: recipes/tensorflow-lite/all/conanfile.py [hunk 4]
```diff
@@ -32,26 +33,18 @@ class TensorflowLiteConan(ConanFile):
...
 
     @property
     def _build_subfolder(self):
-        return "build_subfolder"
+        return "build"
```
Comment (AndreyMlashkin) [inline recipes/tensorflow-lite/all/conanfile.py:47]: Why do you need this change?

File: recipes/tensorflow-lite/all/conanfile.py (hunk 3 above)
Comment (SpaceIm) [reply recipes/tensorflow-lite/all/conanfile.py:43]: to decrease path length for windows

Comment (AndreyMlashkin) [reply recipes/tensorflow-lite/all/conanfile.py:43]: You can force short_paths = True in recipe attributes

Comment (SpaceIm) [reply recipes/tensorflow-lite/all/conanfile.py:43]: don't you see the last commit in this PR?

Comment (AndreyMlashkin) [reply recipes/tensorflow-lite/all/conanfile.py:43]: I see. But do we still need this change then? I would say, it's better to revert it

File: recipes/tensorflow-lite/all/conanfile.py (hunk 4 above)
Comment (uilianries) [reply recipes/tensorflow-lite/all/conanfile.py:47]: Yes. Short paths works, but there are some limitation already commented by SpaceIm. Using src and build its an option on Windows. It's an exception, not the usual case.

**Review by uilianries — APPROVED** [review]

File: recipes/tensorflow-lite/all/conanfile.py (hunk 4 above)
Comment (AndreyMlashkin) [reply recipes/tensorflow-lite/all/conanfile.py:47]: I still don't see the point of changing it, if it works with short paths

**Comment by SpaceIm** [comment]
@ericLemanissier @madebr @SSE4 @jgsogo could you review please, status pending but CI is OK.

**Review by SSE4 — APPROVED** [review]
sure, why not

**Review by madebr — APPROVED** [review]

**Pull request merged**